### PR TITLE
epochs module from osmosis

### PR DESCRIPTION
**Pull request by nguyenvuong1122000**
I add epoch module. It can be useful in future. 
This is cosmos-sdk branch that i replace in go.mod :  [epochs branch](https://github.com/notional-labs/cosmos-sdk)

### CODE REVIEW

**Comment by faddat** [comment]
it's useful right now :)

File: app/app.go [hunk 1]
```diff
@@ -300,6 +306,12 @@ func New(
 	app.StakingKeeper = *stakingKeeper.SetHooks(
 		stakingtypes.NewMultiStakingHooks(app.DistrKeeper.Hooks(), app.SlashingKeeper.Hooks()),
 	)
+	epochsKeeper := epochskeeper.NewKeeper(appCodec, keys[epochstypes.StoreKey])
```
Comment (faddat) [inline app/app.go:309]: Sir, are we missing anything here?

@nguyenvuong1122000 

PS:  Nice work.

**Review by faddat — APPROVED** [review]
passes tests.  Hope to add sims.

File: app/app.go (hunk 1 above)
Comment (nguyenvuong1122000) [reply app/app.go:309]: Oh no, sir. That variable is use to call setHooks func. And now we don't use any Hook.

Comment (nguyenvuong1122000) [reply app/app.go:309]: i use it here :  [[code](https://github.com/notional-labs/dig/blob/91295406d2523bdd0dc7c79d977fc23937574cd9/app/app.go#L310-L313)](https://github.com/notional-labs/dig/blob/91295406d2523bdd0dc7c79d977fc23937574cd9/app/app.go#L310-L313)

**Pull request merged**